bibtex-actions.el

```diff
@@ -234,12 +234,50 @@ If FORCE-REBUILD-CACHE is t, force reloading the cache."
     (bibtex-actions-refresh force-rebuild-cache))
   bibtex-actions--candidates-cache)
 
+(defun bibtex-actions-complete-key-at-point ()
+  "Complete citation key at point.
+
+When inserting '@' in a buffer the capf UI will present user with
+a list of entries, from which they can narrow against title,
+author, etc., and then select one.  This function will then return
+the key 'key', resulting in '@key' at point."
+
+  ;; Activate when user enters '@'.
+  ;;
+  ;; Notes:
+  ;;
+  ;;   - pandoc syntax is [@key], or can also be just @key.
+  ;;   - org syntax is stricter; [cite:see @key1 p23]
+  ;;   - latex uses keys, but sans '@' (I don't want to worry about this ATM though)
+  ;;
+  ;; Maybe this should be constrained further for org?
+  ;;
+  ;; Or, given that pandoc syntax can be used in org context, maybe condition
+  ;; this on 'bibtex-completion-format-citation-functions'?
+  (when (looking-back "@[a-zA-Z0-9]*" 5)
+    (let ((candidates (bibtex-actions--get-candidates)))
+      (list
+       (save-excursion
+         (backward-word)
+         (point))
+       (point)
+       (mapcar #'cdr candidates) :exclusive 'no))))
+
+(defun bibtex-actions-capf ()
+  "My test capf; alt."
+  (when (looking-back "@[a-zA-Z0-9]*" 5)
+    (let* ((candidates (bibtex-actions--get-candidates))
+           ;; FIX
+           (cand (match-string-no-properties 1))
+           (begin (save-excursion (backward-word) (point)))
```

Comment:
This may be wrong now, but I don't actually understand what "save-excursion" does here.

Either way, though, the string can be more than a word.

Reply:
`save-excursion` simply keeps point where it is. The expression moves one word backwards, returns point, and then restores point to wherever it was previously. If you want to move more than one word, you should write your own regexp to match whatever you want (via `looking-back`).

Reply:
If my understanding of  `save-excursion` is correct, this is mostly used for moving around in a buffer performing some action, such as replacing stuff, without driving the user insane, right?